packages/blockchain/test/customConsensus.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { Block, BlockHeader } from '@ethereumjs/block'
+import * as tape from 'tape'
+import Blockchain, { Consensus } from '../src'
+import Common from '@ethereumjs/common'
+
+class fakeConsensus implements Consensus {
+  genesisInit(_genesisBlock: Block): Promise<void> {
+    return new Promise<void>((resolve) => resolve())
+  }
+  setup(): Promise<void> {
+    return new Promise<void>((resolve) => resolve())
+  }
+  validateConsensus(_block: Block): Promise<void> {
+    return new Promise<void>((resolve) => resolve())
+  }
+  validateDifficulty(_header: BlockHeader): Promise<void> {
+    return new Promise<void>((resolve) => resolve())
+  }
+  newBlock(
+    _block: Block,
+    _commonAncestor?: BlockHeader | undefined,
+    _ancientHeaders?: BlockHeader[] | undefined
+  ): Promise<void> {
+    return new Promise<void>((resolve) => resolve())
+  }
+
+  consensusAlgorithm() {
+    return 'fakeConsensus'
+  }
```

Comment:
This should then also be part of the `Consensus` interface.

Reply:
So, and I would then also have a tendency to check this in the Blockchain constructor against the consensus algorithm set in Common and throw if this doesn't match (otherwise this doesn't make things any more consistent and we should in doubt leave completely for now I guess?).

Tests can be adopted in a way that you are just setting this to the `ethash` value and in the interface we can mention in the TSDocs for the `consensusAlgorithm()` method that this will need to match the `Common` one.

Again, I am also still open to leave this completely to simplify things for now. 🙂

Reply:
> This should then also be part of the `Consensus` interface.

Should be addressed in [a41816b](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2002/commits/a41816bd37bc24f7f1e1eaabf08d88b2ec539200)

>So, and I would then also have a tendency to check this in the Blockchain constructor against the consensus algorithm set in Common and throw if this doesn't match (otherwise this doesn't make things any more consistent and we should in doubt leave completely for now I guess?).

If we do this, don't we also have to add in the custom consensus algorithm to `common`?